File: codeforces/67.mishka_and_game/mishka_and_game.cpp
#include <iostream>
/* Author: José Rodolfo (jric2002) */
using namespace std;
int main() {
  unsigned short int n, m, c, m_score = 0, c_score = 0;
  cin >> n;
  while (n--) {
    cin >> m >> c;
    if (m > c) {
      m_score++;
    }
    else if (m < c) {
      c_score++;
    }
    else {
      m_score++;
      c_score++;
    }
  }
  if (m_score > c_score) {
    cout << "Mishka" << endl;
  }
  else if (m_score < c_score) {
    cout << "Chris" << endl;
  }
  else {
    cout << "Friendship is magic!^^" << endl;
  }
  return 0;
}

File: codeforces/67.mishka_and_game_[703A]/mishka_and_game.py
# Author: José Rodolfo (jric2002)
n = int(input())
m_score = 0
c_score = 0
while (n):
  m, c = [int(x) for x in input().split(" ")]
  if (m > c):
    m_score += 1
  elif (m < c):
    c_score += 1
  else:
    m_score += 1
    c_score += 1
  n -= 1
if (m_score > c_score):
  print("Mishka")
elif (m_score < c_score):
  print("Chris")
else:
  print("Friendship is magic!^^")